changed from '.uavcan' to '.dsdl' in DsdlParser

lynet55 opened the pull request:
yolo merge requested right here

lynet55 committed:
changed from '.uavcan' to '.dsdl' in DsdlParser

TheLarsinator commented on RevolveUavcan/Dsdl/DsdlParser.cs:
```diff
@@ -36,14 +36,14 @@ public void ParseAllDirectories()
             foreach (var file in dirs
                 .SelectMany(Directory.GetFiles)
-                .Where(fileName => fileName.Contains(".uavcan")))
+                .Where(fileName => fileName.Contains(".dsdl")))
```
As discussed on Slack, I think a non-breaking change where this become "or .dsdl" would be safer.

lynet55 committed:
added or '.uavcan' for good measure

lynet55 committed:
Update build_and_test.yml

updated from v2 -> v3:
			    - uses: actions/checkout@v3
			    - uses: actions/upload-artifact@v3
			    - uses: actions/upload-artifact@v3

lynet55 committed:
Update build_and_test.yml

Turns out v3 is also depreciated 

https://github.blog/changelog/2024-04-16-deprecation-notice-v3-of-the-artifact-actions/

Trying v4 here.

jkguttormsen commented:
Have you tested it in Analyze with the new dsdl branch and some live data stream? Ask some of the electrical guys, or just replay some old data.

AliceMadama commented:
Nice that you updated the dependency! @jkguttormsen I think they can merge even without testing and then create a new issue/ticket for testing, even if it doesn't work right away it likely is not to be dealt with in this repo anyhow and it won't hurt past logs
